File: JY/8/8.28/체육복.py
def solution(n, lost, reserve):
    ans = 0
    dic = {i:1 for i in range(1,n+1)}
    for i in lost:
        dic[i] -= 1
    for i in reserve:
        dic[i] += 1
    li = [True] + [False]*n
    for i in dic:
        if dic[i]!=0:
            print("지옷있음",i)
            li[i] = True
            dic[i] -= 1
            ans +=1
    
    for i in range(1, n+1):
        if li[i] == False:
            for j in dic:
                if abs(j-i)<=1:
                    if dic[j]>0:
                        print('빌림', i, j,dic[j])
                        dic[j] -= 1
                        ans += 1
                        break
        
                
    return ans

File: YJ/20200904/Greedy/체육복.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int solution(int n, vector<int> lost, vector<int> reserve) {
    int answer = 0;
    vector<int> have_clothes(n + 1, 1);
    
    for (int i = 0; i < lost.size(); i ++) {
        have_clothes[lost[i]] --;
    }
    
    for (int i = 0; i < reserve.size(); i ++) {
        have_clothes[reserve[i]] ++;
    }
        
    for (int i = 1; i <= have_clothes.size(); i ++) {
        if (have_clothes[i] == 2 && i > 1 && have_clothes[i - 1] == 0) {
            have_clothes[i] --;
            have_clothes[i - 1] ++;
        }
        if (have_clothes[i] == 2 && i <= n && have_clothes[i + 1] == 0) {
            have_clothes[i] --;
            have_clothes[i + 1] ++;
        }
    }
    
    for (int i = 1; i < have_clothes.size(); i ++) {
        if (have_clothes[i] > 0) answer ++;
    }
    
    return answer;
}
